### PR TITLE
[Fix 20678] Values cropped on terms and fees page.

### PR DESCRIPTION
**Pull request by DanutGavrus**
### Details
Some values were cropped in On Terms and Fees page. Created a new style to be used here.

### Fixed Issues
$ https://github.com/Expensify/App/issues/20678
PROPOSAL: https://github.com/Expensify/App/issues/20678#issuecomment-1589412694

### Tests
1. Make these change locally:
image
2. Send this message in any chat: "new.expensify.com/enable-payments";
3. From each platform you test, open that chat, and click on that link in order to see the page from this issue;
4. Check the value fields mentioned.
- [X] Verify that no errors appear in the JS console

### Offline tests
Same as Tests Steps.

### QA Steps
1. You need to test from an account which has wallet enbled;
2. Send this message in any chat: "new.expensify.com/enable-payments";
3. From each platform you test, open that chat, and click on that link;
4. Complete the step until you reach the Terms and Fees Page;
5. Check the value fields mentioned. 

- [X] Verify that no errors appear in the JS console

### PR Author Checklist
- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android / native
    - [X] Android / Chrome
    - [X] iOS / native
    - [X] iOS / Safari
    - [X] MacOS / Chrome / Safari
    - [X] MacOS / Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

</details>

<details>
<summary>Desktop</summary>

</details>

<details>
<summary>iOS</summary>

</details>

<details>
<summary>Android</summary>

</details>

### CODE REVIEW

**Comment by DanutGavrus** [comment]
@allroundexperts 
Done adding Screenshots for all platforms. These are done with `lineHeightXXXLarge: getValueUsingPixelRatio(32, 37)` and seem to look good. Let me know if you think we should increase the `lineHeight` though.

File: src/pages/EnablePayments/TermsPage/ShortTermsForm.js [hunk 1]
```diff
@@ -18,7 +20,9 @@ function ShortTermsForm() {
                             <Text style={styles.textLarge}>{Localize.translateLocal('termsStep.monthlyFee')}</Text>
                         </View>
                         <View style={styles.flexRow}>
-                            <Text style={[styles.textHeadline, styles.textXXXLarge]}>{Localize.translateLocal('termsStep.feeAmountZero')}</Text>
+                            <Text style={[styles.textHeadline, styles.textXXXLarge, StyleUtils.getLineHeightStyle(variables.lineHeightXXXLarge)]}>
```
[the remark this answers is not in the document]
Comment (DanutGavrus) [reply src/pages/EnablePayments/TermsPage/ShortTermsForm.js:23]: I did that initially, you may check it [here](https://github.com/Expensify/App/pull/21015/commits/f76ae1e984944089db0480a38f818229b32ede1d), but I changed it after I read these checkboxes:
> - [X] If a new CSS style is added I verified that:
>    - [X] A similar style doesn't already exist
>   - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG))  

Should I go back to using that new style?(if yes, was the naming ok?)

[the remark this answers is not in the document]
Comment (DanutGavrus) [reply src/pages/EnablePayments/TermsPage/ShortTermsForm.js:23]: Could you please check again? The commit does not override the font size of `textHeadline`, or anything else.  

`textXXXLargeWLineHeight` is actually a new style which has the correct `fontSize` & `lineHeight` in it.

[the remark this answers is not in the document]
Comment (DanutGavrus) [reply src/pages/EnablePayments/TermsPage/ShortTermsForm.js]: `textXXXLarge` was already over-riding the line height, so I was thinking you were referring directly to my changes, sorry, my bad. Did another commit & also mergem main with this occasion. Are current [changes](https://github.com/Expensify/App/pull/21015/files) the desired ones?

**Comment by DanutGavrus** [comment]
@allroundexperts I tested it with these changes locally:
image
Then, open any chat, send this message: "new.expensify.com/enable-payments", and click on if from any platform in order to open that page.

**Comment by DanutGavrus** [comment]
@allroundexperts
> Can you update the test steps and mention that we need an account with wallet enabled in order to test this?
Both Tests & QA Tests updated accordingly.

**Review by allroundexperts — APPROVED** [review]
Looks good to me!

Waiting on @shawnborton to verify if line height is correct in the screenshots.

**Comment by shawnborton** [comment]
This generally looks good, but for this:
image


Can we add some margin in between the in-network and out-of-network? Maybe just add like 12-16px so there is a bit of a gap.

**Comment by DanutGavrus** [comment]
<details>
<summary>A - 8px</summary>

</details>

<details>
<summary>B - 12px</summary>

</details>

<details>
<summary>C - 16px</summary>

</details>

<details>
<summary>D - 32px</summary>

</details>

@shawnborton A, B, C, D, or other?

**Comment by shawnborton** [comment]
C - 16px looks good to me!

**Comment by DanutGavrus** [comment]
@shawnborton Done
@allroundexperts I think we are good to go now, I've also merged main again

**Review by allroundexperts — APPROVED** [review]

**Comment by DanutGavrus** [comment]
@deetergp bump for final review 👀

**Review by deetergp — APPROVED** [review]
Looks good, thanks for the PR!

**Pull request merged**